### PR TITLE
JNG-5294 Refactor FiltersModel tests

### PR DESCRIPTION
**Pull request by gaborflorian**
<table>
<td>
  <a href="https://blackbelt.atlassian.net/browse/JNG-5294" title="JNG-5294" target="_blank">TaskJNG-5294</a>  Refactor judo-runtime-core-jsl FiltersModel tests
  </td></table>
  <br />

### CODE REVIEW

File: judo-runtime-core-jsl-itest/models/FilterModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/FiltersTest.java [hunk 1]
```diff
@@ -218,108 +218,22 @@ public void testIntegerNumberFilter() {
...
-        assertEquals(2, lessOrEqual.size());
-
-        MyEntityWithOptionalFields less = myEntityWithOptionalFieldsDao
-                .query()
-                .filterByIntegerAttr(NumberFilter.lessThan(INTEGER_2))
-                .selectOne()
-                .get();
-
-        assertEquals(entity1.identifier(), less.identifier());
+        costumFilter(entity1, "this.integerAttr >= 1 and this.integerAttr < 2");
```
Comment (noherczeg) [inline judo-runtime-core-jsl-itest/models/FilterModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/FiltersTest.java]: costum?

File: judo-runtime-core-jsl-itest/models/FilterModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/FiltersTest.java [hunk 2]
```diff
@@ -2519,4 +1094,42 @@ void testNavigationOnFilter() {
...
+    private void costumFilterList(int expected, String customFilter) {
+        List<MyEntityWithOptionalFields> greaterOrEqualsByStringFilter = myEntityWithOptionalFieldsDao
+                .query()
+                .filterBy(customFilter)
+                .selectList();
+
+        assertEquals(expected, greaterOrEqualsByStringFilter.size());
+    }
+
+    private void costumFilter(MyEntityWithOptionalFields entity, String customFilter) {
```
Comment (noherczeg) [inline judo-runtime-core-jsl-itest/models/FilterModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/FiltersTest.java]: name should be `assertEntityMatchesFilter`.

File: judo-runtime-core-jsl-itest/models/FilterModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/FiltersTest.java [hunk 3]
```diff
@@ -2519,4 +1094,42 @@ void testNavigationOnFilter() {
...
+    private void filterByNameList(int expected, String integerAttr, Filter filter1) {
+        List<MyEntityWithOptionalFields> equalsByName = myEntityWithOptionalFieldsDao
+                .query()
+                .filterByName(integerAttr, filter1)
+                .selectList();
+
+        assertEquals(expected, equalsByName.size());
+    }
+
+    private void costumFilterList(int expected, String customFilter) {
```
Comment (noherczeg) [inline judo-runtime-core-jsl-itest/models/FilterModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/FiltersTest.java]: Name should be `assertFilterMatchesNumberOfResults`

File: judo-runtime-core-jsl-itest/models/FilterModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/FiltersTest.java [hunk 4]
```diff
@@ -2519,4 +1094,42 @@ void testNavigationOnFilter() {
...
+        MyEntityWithOptionalFields equalsByName = myEntityWithOptionalFieldsDao
+                .query()
+                .filterByName(integerAttr, filter)
+                .selectOne()
+                .get();
+
+        assertEquals(entity.identifier(), equalsByName.identifier());
+    }
+
+    private void filterByNameList(int expected, String integerAttr, Filter filter1) {
```
Comment (noherczeg) [inline judo-runtime-core-jsl-itest/models/FilterModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/FiltersTest.java]: Name should be `assertFilterByNameMatchesNumberOfResults`

File: judo-runtime-core-jsl-itest/models/FilterModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/FiltersTest.java [hunk 5]
```diff
@@ -2519,4 +1094,42 @@ void testNavigationOnFilter() {
 
     }
 
+    private void filterByName(MyEntityWithOptionalFields entity, String integerAttr, Filter filter) {
```
Comment (noherczeg) [inline judo-runtime-core-jsl-itest/models/FilterModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/FiltersTest.java]: Name should be `assertFilterByNameMatchesEntity`

Comment (noherczeg) [inline judo-runtime-core-jsl-itest/models/FilterModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/FiltersTest.java]: integerAttr?

File: judo-runtime-core-jsl-itest/models/FilterModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/transfer/MappedTransferFiltersTest.java [hunk 6]
```diff
@@ -2508,4 +1106,42 @@ void testNavigationOnFilterOnTransfer() {
...
+    private void costumFilterList(int expected, String customFilter) {
+        List<TransferMyEntityWithOptionalFields> greaterOrEqualsByStringFilter = transferMyEntityWithOptionalFieldsDao
+                .query()
+                .filterBy(customFilter)
+                .selectList();
+
+        assertEquals(expected, greaterOrEqualsByStringFilter.size());
+    }
+
+    private void costumFilter(TransferMyEntityWithOptionalFields entity, String customFilter) {
```
Comment (noherczeg) [inline judo-runtime-core-jsl-itest/models/FilterModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/transfer/MappedTransferFiltersTest.java]: Name should be `assertEntityMatchesFilter`

File: judo-runtime-core-jsl-itest/models/FilterModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/transfer/MappedTransferFiltersTest.java [hunk 7]
```diff
@@ -2508,4 +1106,42 @@ void testNavigationOnFilterOnTransfer() {
...
+    private void filterByNameList(int expected, String integerAttr, Filter filter1) {
+        List<TransferMyEntityWithOptionalFields> equalsByName = transferMyEntityWithOptionalFieldsDao
+                .query()
+                .filterByName(integerAttr, filter1)
+                .selectList();
+
+        assertEquals(expected, equalsByName.size());
+    }
+
+    private void costumFilterList(int expected, String customFilter) {
```
Comment (noherczeg) [inline judo-runtime-core-jsl-itest/models/FilterModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/transfer/MappedTransferFiltersTest.java]: Name should be `assertFilterMatchesNumberOfResults`

File: judo-runtime-core-jsl-itest/models/FilterModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/transfer/MappedTransferFiltersTest.java [hunk 8]
```diff
@@ -2508,4 +1106,42 @@ void testNavigationOnFilterOnTransfer() {
 
     }
 
+    private void filterByName(TransferMyEntityWithOptionalFields entity, String integerAttr, Filter filter) {
```
Comment (noherczeg) [inline judo-runtime-core-jsl-itest/models/FilterModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/transfer/MappedTransferFiltersTest.java]: integerAttr???

Comment (noherczeg) [inline judo-runtime-core-jsl-itest/models/FilterModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/transfer/MappedTransferFiltersTest.java]: Name should be `assertFilterByNameMatchesEntity`

**Review by noherczeg — CHANGES_REQUESTED** [review]

File: judo-runtime-core-jsl-itest/models/FilterModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/transfer/MappedTransferFiltersTest.java (hunk 8 above)
Comment (gaborflorian) [reply judo-runtime-core-jsl-itest/models/FilterModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/transfer/MappedTransferFiltersTest.java]: [500a85f](https://github.com/BlackBeltTechnology/judo-runtime-core-jsl/pull/260/commits/500a85fad8e240281a5160a7acc1c88358a0bb5c)

File: judo-runtime-core-jsl-itest/models/FilterModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/transfer/MappedTransferFiltersTest.java (hunk 7 above)
Comment (gaborflorian) [reply judo-runtime-core-jsl-itest/models/FilterModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/transfer/MappedTransferFiltersTest.java]: [500a85f](https://github.com/BlackBeltTechnology/judo-runtime-core-jsl/pull/260/commits/500a85fad8e240281a5160a7acc1c88358a0bb5c)

File: judo-runtime-core-jsl-itest/models/FilterModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/transfer/MappedTransferFiltersTest.java (hunk 6 above)
Comment (gaborflorian) [reply judo-runtime-core-jsl-itest/models/FilterModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/transfer/MappedTransferFiltersTest.java]: [500a85f](https://github.com/BlackBeltTechnology/judo-runtime-core-jsl/pull/260/commits/500a85fad8e240281a5160a7acc1c88358a0bb5c)

File: judo-runtime-core-jsl-itest/models/FilterModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/FiltersTest.java (hunk 5 above)
Comment (gaborflorian) [reply judo-runtime-core-jsl-itest/models/FilterModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/FiltersTest.java]: [500a85f](https://github.com/BlackBeltTechnology/judo-runtime-core-jsl/pull/260/commits/500a85fad8e240281a5160a7acc1c88358a0bb5c)

File: judo-runtime-core-jsl-itest/models/FilterModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/transfer/MappedTransferFiltersTest.java (hunk 8 above)
Comment (gaborflorian) [reply judo-runtime-core-jsl-itest/models/FilterModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/transfer/MappedTransferFiltersTest.java]: [2250748](https://github.com/BlackBeltTechnology/judo-runtime-core-jsl/pull/260/commits/22507482b0d1eae7e5d7b79c2ccb799a6acb9330)

File: judo-runtime-core-jsl-itest/models/FilterModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/FiltersTest.java (hunk 5 above)
Comment (gaborflorian) [reply judo-runtime-core-jsl-itest/models/FilterModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/FiltersTest.java]: [2250748](https://github.com/BlackBeltTechnology/judo-runtime-core-jsl/pull/260/commits/22507482b0d1eae7e5d7b79c2ccb799a6acb9330)

File: judo-runtime-core-jsl-itest/models/FilterModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/transfer/MappedTransferFiltersTest.java (hunk 8 above)
Comment (gaborflorian) [reply judo-runtime-core-jsl-itest/models/FilterModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/transfer/MappedTransferFiltersTest.java]: [500a85f](https://github.com/BlackBeltTechnology/judo-runtime-core-jsl/pull/260/commits/500a85fad8e240281a5160a7acc1c88358a0bb5c)

File: judo-runtime-core-jsl-itest/models/FilterModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/FiltersTest.java (hunk 5 above)
Comment (gaborflorian) [reply judo-runtime-core-jsl-itest/models/FilterModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/FiltersTest.java]: [500a85f](https://github.com/BlackBeltTechnology/judo-runtime-core-jsl/pull/260/commits/500a85fad8e240281a5160a7acc1c88358a0bb5c)

File: judo-runtime-core-jsl-itest/models/FilterModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/FiltersTest.java (hunk 4 above)
Comment (gaborflorian) [reply judo-runtime-core-jsl-itest/models/FilterModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/FiltersTest.java]: [500a85f](https://github.com/BlackBeltTechnology/judo-runtime-core-jsl/pull/260/commits/500a85fad8e240281a5160a7acc1c88358a0bb5c)

File: judo-runtime-core-jsl-itest/models/FilterModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/FiltersTest.java (hunk 3 above)
Comment (gaborflorian) [reply judo-runtime-core-jsl-itest/models/FilterModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/FiltersTest.java]: [500a85f](https://github.com/BlackBeltTechnology/judo-runtime-core-jsl/pull/260/commits/500a85fad8e240281a5160a7acc1c88358a0bb5c)

File: judo-runtime-core-jsl-itest/models/FilterModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/FiltersTest.java (hunk 2 above)
Comment (gaborflorian) [reply judo-runtime-core-jsl-itest/models/FilterModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/FiltersTest.java]: [500a85f](https://github.com/BlackBeltTechnology/judo-runtime-core-jsl/pull/260/commits/500a85fad8e240281a5160a7acc1c88358a0bb5c)

File: judo-runtime-core-jsl-itest/models/FilterModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/FiltersTest.java (hunk 1 above)
Comment (gaborflorian) [reply judo-runtime-core-jsl-itest/models/FilterModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/FiltersTest.java]: [500a85f](https://github.com/BlackBeltTechnology/judo-runtime-core-jsl/pull/260/commits/500a85fad8e240281a5160a7acc1c88358a0bb5c)

**Review by robertcsakany — APPROVED** [review]

**Pull request merged**